wazo_test_helpers/asset_launching_test_case.py

```diff
@@ -346,25 +379,87 @@ def _maybe_dump_docker_logs(cls):
             )
             AssetLaunchingTestCase.log_dir = os.getenv(
-                'WAZO_TEST_DOCKER_LOGS_DIR', default_logging_dir
+                'WAZO_TEST_DOCKER_LOGS_DIR', f'/tmp/wazo-integration-{random_strings}'
             )
             if not os.path.exists(AssetLaunchingTestCase.log_dir):
                 os.makedirs(AssetLaunchingTestCase.log_dir, mode=0o755)
-        return AssetLaunchingTestCase.log_dir
+        return str(AssetLaunchingTestCase.log_dir)
 
 
-def _run_cmd(cmd, stderr=True):
-    logger.debug('%s', cmd)
-    stderr = subprocess.STDOUT if stderr else None
+class AssetLaunchingTestCase(AbstractAssetLaunchingHelper, unittest.TestCase):
+    @classmethod
+    def setUpClass(cls) -> None:
+        cls.launch_service_with_asset()
+
+    @classmethod
+    def tearDownClass(cls) -> None:
+        cls.stop_service_with_asset()
+
+    @classmethod
+    def is_managing_containers(cls) -> bool:
+        warnings.warn(
+            "Method `is_managing_containers` was removed, please use "
+            "`get_container_management_enabled` or migrate to fixtures.",
+            DeprecationWarning,
+            stacklevel=2,
+        )
+        return get_container_management_enabled()
 
-    completed_process = subprocess.run(cmd, stdout=subprocess.PIPE, stderr=stderr)
+
+def make_asset_fixture(
+    asset_class: type[AbstractAssetLaunchingHelper],
+) -> Callable[[], Generator[type[AbstractAssetLaunchingHelper], None, None]]:
+    """
+    Helper to create asset loading fixtures for Pytest in conftest.py or test files.
+
+    Usage:
+    ```python
+    class MyAssetClass(AbstractAssetLaunchingHelper):
+        assets_root = Path(__file__).parent / '..' / '..' / 'assets'
+        asset = 'asset-name'
+        service = 'my-service'
+
+    # Note: assignment is important to be detected by pytest.
+    asset_name_asset = make_asset_fixture(MyAssetClass)
+
+    # For test methods
+    def test_my_test(asset_name_asset: MyAssetClass) -> None:
+        asset_name_asset.service_status()  # Containers are up, and you can call methods on
+
+    # For whole classes:
+    @pytest.mark.usefixtures('asset_name_asset')
+    class TestExample(TestCase):
+        def test_my_test(self) -> None:
+            self.asset_name_asset.service_status()  # Containers are up, and you can call methods
+    """
+    asset_name = asset_class.asset.replace('-', '_')
+
+    @pytest.fixture(name=f'{asset_name}_asset', scope='session')
+    def fixture() -> Generator[type[AbstractAssetLaunchingHelper], None, None]:
+        if (
+            not issubclass(asset_class, AbstractAssetLaunchingHelper)
+            or asset_class is AbstractAssetLaunchingHelper
+        ):
+            raise TypeError(
+                'You must subclass `AbstractAssetLaunchingHelper` and pass that class instead.'
+            )
+        asset_class.launch_service_with_asset()
+        yield asset_class
+        asset_class.stop_service_with_asset()
```

Comment:
Should the `yield` be wrapped by a `try..finally` to be sure to execute stop if something goes wrong?